src/coreclr/jit/compiler.cpp

```diff
@@ -3069,8 +3069,17 @@ void Compiler::compInitOptions(JitFlags* jitFlags)
         }
         s_pJitFunctionFileInitialized = true;
     }
-
-#endif // DEBUG
+#else  // DEBUG
+    if (!JitConfig.JitDisasm().isEmpty())
+    {
+        const char* methodName = info.compCompHnd->getMethodName(info.compMethodHnd, nullptr);
+        const char* className  = info.compCompHnd->getClassName(info.compClassHnd);
+        if (JitConfig.JitDisasm().contains(methodName, className, &info.compMethodInfo->args))
```

Comment:
Could you please delete the support for NgenDisasm (and all other `NGen*`) from the Debug block above so that there is no behavior mismatch between checked and release JITs?

Reply:
Actually, it can be done as separate PR so that this one stay focused on enabling the disasm in Release: https://github.com/dotnet/runtime/pull/73383

Reply:
Yes, I've done Ngen* removal locally and it was pretty verbose so I didn't want to distract code reviewers 🙂